internal/image/history.go

```diff
@@ -89,25 +93,91 @@ func (h ImageHistory) load(ctx context.Context, name reference.Named, digest dig
 		h.byName[key] = bucket
 	}
 
+	bucket.entries = append(bucket.entries, entry)
+	if entry.After(bucket.mostRecent.CheckpointID) {
+		bucket.mostRecent = entry
+	}
+}
+
+func (h ImageHistory) load(
```

Comment:
is this replicating logic from `loadFromEntry`? maybe this should just assemble all the inputs into a `historyEntry` and call `loadFromEntry` on it?